lib/index.js

```diff
@@ -206,25 +224,23 @@ gcloud.dns = DNS;
  * @example
  * var gcloud = require('gcloud');
  * var pubsub = gcloud.pubsub({
- *   projectId: 'project-id',
+ *   projectId: 'grape-spaceship-123',
  *   keyFilename: '/path/to/keyfile.json'
  * });
  */
-gcloud.pubsub = function(config) {
-  return new PubSub(config);
-};
+gcloud.pubsub = PubSub;
```

Comment:
Awkwardly, `pubsub` and `search` were doing more work than necessary. I adopted the system we use for our other APIs, and we now do auto-instantiation in the constructor, e.g.:

```js
function PubSub(options) {
  if (!(this instanceof PubSub)) {
    return new PubSub(options);
  }
  // ...
}
```